lasagne/layers/input.py

```diff
@@ -49,6 +49,12 @@ class InputLayer(Layer):
     """
     def __init__(self, shape, input_var=None, name=None, **kwargs):
         self.shape = shape
+        if any(d is not None and d <= 0 for d in self.shape):
+            raise ValueError((
+                "Cannot create InputLayer with a non-positive shape "
+                "dimension. shape=%r, self.name=%s") % (
+                    self.shape, str(name)))
```

Comment:
Can you please just follow the code from the base Layer constructor? It uses `self.name=%r` and uses `name` directly (not wrapped in `str()` -- note that this doesn't make a difference when using `%s` anyway, it implicitly calls `str()` on the argument to be formatted). Cheers!